### ISSUE TITLE
ViewScene shows unexpected values when clipping

### ISSUE THREAD

**Issue by Sumientra** [SCIInstitute/SCIRun#2213]
**Describe the bug**
I am visualizing a tet mesh with data (electric field) on the elements. When I clip and slide through it, there are unexpected values / outliers. When I rotate they disappear. It looks like the value on the inside and outside of the tet is different. Hard to explain; [here](https://drive.google.com/file/d/1tJbOeKs3DZ3nod6Yk-9LKTaG_dQdsAWc/view?usp=sharing) is a video. 

**Version**
v5.0-beta.Y-172-g1d75a9c3f
v5.0-beta.Y.RC1+Qt5.14.1

**Desktop (please complete the following information):**
macOS 11.1
macOS 10.15.7

**Comment by dcwhite** [comment]
Is it the blue stuff I circled in red here?
![image](https://user-images.githubusercontent.com/1399383/105395968-93784e80-5bdc-11eb-8438-21dfd3118ddd.png)

**Comment by Sumientra** [comment]
Yes that's it. It disappears upon rotating.

**Comment by jessdtate** [comment]
this looks like it could be degenerate or sliver tets.  Did you confirm that the visualization is different from the data on the mesh?  

It would be helpful to have an example net with data to investigate.

**Comment by Sumientra** [comment]
The problem is in the ViewScene clipping function. Here are two images of the same model.

1: Clipped the mesh using ClipFieldByFunction, then visualized -> good
![clip field](https://user-images.githubusercontent.com/20671873/105669556-880f7680-5ead-11eb-821d-dd84df98b0b4.png)

2: Visualized complete mesh, then clipped using ViewScene -> bad
![clip vis](https://user-images.githubusercontent.com/20671873/105669671-b8efab80-5ead-11eb-924e-4abb8950cd91.png)

**Comment by Sumientra** [comment]
[Here](https://drive.google.com/drive/folders/17w5n1x7EBQEn5AHe1HnNYRbYLRIJYYpm?usp=sharing) is the network and data that produced these images.

**Comment by dcwhite** [comment]
I feel like the ViewScene clipping function will just be an approximation to whatever the real clipped field will look like, so this is "expected" in the sense that they aren't required to match. ViewScene is doing an off-the-cuff clipping job (a SuperCuts haircut rather than an expensive salon haircut? @jessdtate can weigh in on that analogy). However, we can look into improving the VS clipping algorithm so that it is closer to the mesh clipping result.

**Comment by jessdtate** [comment]
There are some approximations happening with the clipping, but these colors aren't supposed to be happening.  It seems like the renderer is mapping the wrong colors to the back side of the faces.  This seems to be data specific as I can only produce the effect with this example, including the UQ @Sumientra shared previously.  @Sumientra, how many datasets do you see this with?  

Since it is data specific, there are some work arounds.  getting just the boundary fixes the rendering, as does using ClipVolumebyIsoValue.  There are probably others too.

**Comment by jessdtate** [comment]
I think I just found the problem.  The data in the example are vectors, which I suppose we didn't test with the clipping.  The vectors seems to be interfering with the color lookup.  It still works with tri surfs, but I think we need to test a few mesh types with vectors and tensor data to see how they respond.  I wonder if @RubioJr9 might have some time to look at it.

**Comment by Sumientra** [comment]
Yes this is it! I just took the norm before visualizing and that solves it.

**Comment by jessdtate** [comment]
Here is a simpler example that shows the problem with vector data
[test_latvol.srn5.zip](https://github.com/SCIInstitute/SCIRun/files/5896465/test_latvol.srn5.zip)

**Comment by jessdtate** [comment]
looks like tensors has the same problem.  
[test_latvol_tensor.srn5.zip](https://github.com/SCIInstitute/SCIRun/files/5896950/test_latvol_tensor.srn5.zip)

**Comment by jessdtate** [comment]
Another point to note is that the bug only happens with constant mapping on the elements.

**Comment by dcwhite** [comment]
**Analysis (2026-06-13)**

ViewScene clipping with unexpected colors for vector/tensor data. Root cause identified in comments: when clipping is enabled with element-centered vector or tensor data, the color lookup uses the wrong data (vectors interfere with scalar colormap lookup). Confirmed with test network. Fix: in the clipping shader or geometry generation, ensure scalar colormap lookup uses only the mapped scalar value, not raw vector components. Affects vector+tensor, constant (element-centered) mapping only.
